tensorly/regression/cp_plsr.py

```diff
@@ -177,15 +230,30 @@ def predict(self, X):
             raise ValueError(
                 f"Training X has shape {self.X_shape_}, while the new X has shape {T.shape(X)}"
             )
+        X = T.copy(X)
         X -= self.X_mean_
-        factors_kr = khatri_rao(self.X_factors, skip_matrix=0)
-        unfolded = unfold(X, 0)
-        scores = T.lstsq(factors_kr, T.transpose(unfolded))[0]  # = Tnew
-        estimators = T.lstsq(self.X_factors[0], self.Y_factors[0])[0]
-        return (
-            T.dot(
-                T.dot(T.transpose(scores), estimators), T.transpose(self.Y_factors[1])
+        X_projection = T.zeros((T.shape(X)[0], self.n_components), **T.context(X))
+        for component in range(self.n_components):
+            X_projection = T.index_update(
+                X_projection,
+                T.index[:, component],
+                multi_mode_dot(
+                    X,
+                    [factor[:, component] for factor in self.X_factors[1:]],
+                    range(1, T.ndim(X)),
+                ),
             )
+            X -= cp_to_tensor((
+                    None,
+                    [T.reshape(X_projection[:, component], (-1, 1))]
+                    + [
+                        T.reshape(factor[:, component], (-1, 1))
+                        for factor in self.X_factors[1:]
+                    ],
```

Comment:
Same here, we could just do the tl.tenalg.outer if it proves faster? Might be more readable too?